tick/stateful/eval_function_node.go

```diff
@@ -187,6 +216,12 @@ func eval(n NodeEvaluator, scope *Scope, executionState ExecutionState) (interfa
 		return n.EvalTime(scope, executionState)
 	case ast.TDuration:
 		return n.EvalDuration(scope, executionState)
+	case ast.TMissing:
```

Comment:
There could be another way of doing it, this is just what seemed most clear to me.

Here's my reasoning (in a some what deductive order):

Typically, in tickscript if a field or tag doesn't exist, when the scope is field, an error is returned. So the first thing I did was change this so that instead, a missing value was set.
```
		if !isFieldExists && !isTagExists {
			if !vars.Has(refVariableName) {
				var missingValue interface{} = &ast.Missing{}
				vars.Set(refVariableName, missingValue)
				//return fmt.Errorf("no field or tag exists for %s", refVariableName)
			}
		}
```
https://github.com/influxdata/kapacitor/blob/c55df2732113e3e82ea72f4dd62367b4b3fd2499/expr.go#L54-L55

The function `isPresent` must be passed some value to check presence against (the lack of a value isn't something you can check). This value is a reference and must have a value distinct from all other types of values (so that `isPresent` can properly distinguish between present and non-present values). 

At some point, this reference must be evaluated to a value as it must be passed to the `isPresent` function.
```
	for i, argEvaluator := range n.argsEvaluators {
		value, err := eval(argEvaluator, scope, executionState)
		if err != nil {
			return nil, fmt.Errorf("Failed to handle %v argument: %v", i+1, err)
		}

		args = append(args, value)
	}
```
https://github.com/influxdata/kapacitor/blob/c55df2732113e3e82ea72f4dd62367b4b3fd2499/tick/stateful/eval_function_node.go#L71

Since we have to evaluate the reference to a missing value, we need an `EvalMissing` function on `NodeEvaluator` (because of the function signature of `eval`).
```
func eval(n NodeEvaluator, scope *Scope, executionState ExecutionState)
```
https://github.com/influxdata/kapacitor/blob/c55df2732113e3e82ea72f4dd62367b4b3fd2499/tick/stateful/eval_function_node.go#L198